UnityProject/Assets/Scripts/Managers/MatrixManager.cs

```diff
@@ -11,341 +11,356 @@
-		if (component && component.netId != NetworkInstanceId.Invalid && !component.netId.IsEmpty())
-		{
-			netId = component.netId;
-		}
-
-		if (componentInParent && componentInParent.netId != NetworkInstanceId.Invalid && !componentInParent.netId.IsEmpty())
-		{
-			netId = componentInParent.netId;
-		}
-
-		if (netId == NetworkInstanceId.Invalid)
-		{
-			Logger.LogWarning($"Invalid NetID for matrix {matrix.gameObject.name}!", Category.Matrix);
-		}
-
-		return netId;
-	}
+    public int Id;
+    public Matrix Matrix;
+    public MetaTileMap MetaTileMap;
+    public GameObject GameObject;
+
+    public Vector3Int InitialOffset;
+
+    public Vector3Int Offset => GetOffset();
+    public Vector3Int GetOffset(MatrixState state = default(MatrixState))
+    {
+        if (!MatrixMove)
+        {
+            return InitialOffset;
+        }
+        if (state.Equals(default(MatrixState)))
+        {
+            state = MatrixMove.ClientState;
+        }
+        return InitialOffset + (Vector3Int.RoundToInt(state.Position) - MatrixMove.InitialPos);
+    }
+
+    public MatrixMove MatrixMove;
+
+    private NetworkInstanceId netId;
+
+    public NetworkInstanceId NetId
+    {
+        get
+        {
+            //late init, because server is not yet up during InitMatrices()
+            if (netId.IsEmpty() || netId == NetworkInstanceId.Invalid)
+            {
+                netId = getNetId(Matrix);
+            }
+
+            return netId;
+        }
+    }
+
+    /// Null object
+    public static readonly MatrixInfo Invalid = new MatrixInfo { Id = -1 };
+
+    public override string ToString()
+    {
+        return Equals(Invalid)
+            ? "[Invalid matrix]"
+            : $"[({Id}){GameObject.name},offset={Offset},pivot={MatrixMove?.Pivot},state={MatrixMove?.State},netId={NetId}]";
+    }
+
+    ///Figuring out netId. NetworkIdentity is located on the pivot (parent) gameObject for MatrixMove-equipped matrices
+    private static NetworkInstanceId getNetId(Matrix matrix)
+    {
+        var netId = NetworkInstanceId.Invalid;
+        if (!matrix)
+        {
+            return netId;
+        }
+
+        NetworkIdentity component = matrix.gameObject.GetComponentInParent<NetworkIdentity>();
+        NetworkIdentity componentInParent = matrix.gameObject.GetComponentInParent<NetworkIdentity>();
+        if (component && component.netId != NetworkInstanceId.Invalid && !component.netId.IsEmpty())
+        {
+            netId = component.netId;
+        }
+
+        if (componentInParent && componentInParent.netId != NetworkInstanceId.Invalid && !componentInParent.netId.IsEmpty())
+        {
+            netId = componentInParent.netId;
+        }
+
+        if (netId == NetworkInstanceId.Invalid)
+        {
+            Logger.LogWarning($"Invalid NetID for matrix {matrix.gameObject.name}!", Category.Matrix);
+        }
+
+        return netId;
+    }
```

Comment:
What was changed here? I'm not reading through the diff to find out as it was reformatted